xbmc/FileItem.cpp

```diff
@@ -1933,6 +1933,15 @@ void CFileItem::SetDynPath(const std::string &path)
   m_strDynPath = path;
 }
 
+const std::string& CFileItem::GetPluginPath() const
```

Comment:
I don't like the name of this method. Dynpath always take precedence over path, the resolve instruction just takes whatever is in path and resolves it to dynpath. However, this method might return the original path. Maybe something like GetOriginalPluginPath. Also I don't think this belongs to CFileItem as it is specific to an item type (plugin). Can you move this to URIUtils or a similar class? Anything that avoids stuffing new methods in CFileItem is better.

Reply:
done